src/audio.rs

```diff
@@ -128,7 +128,14 @@ impl From<f32> for S24 {
 
 impl From<S24> for f32 {
     fn from(x: S24) -> f32 {
-        ((x.0 ^ S24_SIGN) - S24_SIGN) as f32 * S24_TO_F32_SCALE
```

Comment:
This was giving me a "attempted to subtract with overflow" on one of the examples. I am assuming that the more verbose version that I change this to will be optimised in compilation.

Reply:
These two are a little different. I did some testing in godbolt, and it looks like the compiler, at least for x86, has a harder time with the branch.

You can read a bit more about this [particular technique here](https://graphics.stanford.edu/~seander/bithacks.html#VariableSignExtend), among other places. In short, it's a sign-extension technique.

It should only ever overflow if the top eight bits of the incoming value are non-zero. When reading from the SAI DMA stream, this should not be possible as far as I can tell.

To prevent any panics while maintaining performance, we can just do the arithmetic shift trick:
```rs
((x.0 << 8) >> 8) as f32 * S24_TO_F32_SCALE
```

In general, though, it's a little concerning that those bits aren't zero. What was the example that caused the failure?

Reply:
The delay example is the one in question.